src/app/routes/utils/constructPageFetchUrl/index.ts

```diff
@@ -0,0 +1,94 @@
+import Url from 'url-parse';
+import pipe from 'ramda/src/pipe';
+import getEnvironment from '#app/routes/utils/getEnvironment';
+import { getUrlPath } from '../../../lib/utilities/urlParser';
+import handleError from '../handleError';
+import { Services, Variants } from '../../../models/types/global';
+
+interface UrlConstructParams {
+  pathname: string;
+  pageType: 'article' | 'cpsAsset' | 'topic' | 'live';
+  service: Services;
+  variant?: Variants;
+  page?: string;
+}
+
+const removeAmp = (path: string) => path.split('.')[0];
+const getArticleId = (path: string) => path.match(/(c[a-zA-Z0-9]{10}o)/)?.[1];
+const getTipoId = (path: string) => path.match(/(c[a-zA-Z0-9]{10}t)/)?.[1];
+const getCpsId = (path: string) => path;
+
+const getId = (pageType: string) => {
```

Comment:
I am wondering whether we should include the getId functionality here - I worry that this may become a little bit more tricky to maintain in future. Should it be the responsibility of the thing which calls this to pass in the ID? If there are commonalities in deriving ID from the path, could that be added a library elsewhere, and called in the getInitialData for the respective page type instead? e.g. move getArticleId and getCpsId to a utility and call it when you need it?

Reply:
I mentioned in the thread a bit on this. During the CAF spike, I moved this logic into the BFF itself: https://github.com/bbc/fabl-modules/pull/3740/commits/aaccfebdbe5526c583698b8b210751eb0c520306 

Simorgh would just pass up the raw URL and then the BFF would handle parsing it out. Each route would handle parsing itself https://github.com/bbc/fabl-modules/pull/3740/commits/aaccfebdbe5526c583698b8b210751eb0c520306#diff-957e9d658e3770616a7890b3aafd68577fc8868201d3b554852c89a62beab942R112-R113

This was something of a necessity for the spike work, as the BFF needed to know if the page was an AMP page or not, so I got around passing in another query param by just passing in the raw URL. May or may not be the best approach, but at least moves more of this logic out of Simorgh.

Reply:
What do we think is best then? Do we encapsulate this in another ticket to migrate the validation logic over to the BFF? Or shall we move back these functions to co-locate them and extract out the common ones like `removeAmp`, into it's own utility? I don't want to bloat this PR too much as it was just meant to be for hitting the new live page endpoint

Reply:
I don't want to hold you up on this, I think what you have is good, and it works, and maybe worth considering a migration over to the BFF at a later stage? Create a ticket and chuck it in the backlog 😝

Reply:
Yea ditto, what you have looks good and we can take a look at moving as much as we can to the BFF at a later stage 👍🏻

Reply:
One thought about moving the ID functionality to the BFF is the fact that the homepage doesn't have an ID which is parsable from the URL. So we would always need to pass it in from Simorgh. But that's a 🐮 point right now, just worth remembering if/when we implement this logic at a later date.